src/devices/x10_sec.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * X10 Security sensor decoding for rtl_433
+ *
+ * Based on code provided by Willi 'wherzig' in issue #30 (2014-04-21)
+ * https://github.com/merbanan/rtl_433/issues/30
+ * Danke, Willi
+ *
+ * Tested with American sensors operating at 310 MHz
+ * e.g., rtl_433 -f 310.558M
+ *
+ * This is pretty rudimentary, and I bet the byte value decoding, based
+ * on limited observations, doesn't take into account bits that might
+ * be set to indicate something like a low battery condition.
+ *
+ * Copyright (C) 2018 Anthony Kava
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ */
+
+#include "rtl_433.h"
+#include "data.h"
+#include "util.h"
+
+static int x10_sec_callback(bitbuffer_t *bitbuffer) {
+    bitrow_t *bb = bitbuffer->bb;
+    data_t *data;
+    uint16_t r;                          /* a row index              */
+    uint8_t *b;                          /* bits of a row            */
+    uint8_t test_a;                      /* test 1/2 for message     */
+    uint8_t test_b;                      /* test 2/2 for message     */
+    char *event_str = "UNKNOWN";         /* human-readable event     */
+    char x10_id_str[5] = "\0";           /* string showing hex value */
+    char x10_code_str[5] = "\0";         /* string showing hex value */
+    char time_str[LOCAL_TIME_BUFLEN];
+
+    for (r=0; r < bitbuffer->num_rows; ++r) {
+        b = bitbuffer->bb[r];
+
+        /* looking for five bytes */
+        if (bitbuffer->bits_per_row[r] < 40) {
+            continue;
+        }
+
+        /* validate what we received ...
+         * used intermediate variables test_a and test_b to suppress
+         * warnings generated when compiling using:
+         *
+         *   if((b[0]^0x0f)==b[1] && (b[2]^0xff)==b[3])
+         */
+        test_a = b[0] ^ 0x0f;
+        test_b = b[2] ^ 0xff;
+        if (test_a != b[1] || test_b != b[3]) {
+            continue;
+        } else {
```

Comment:
you don't need this `else`, which reduces the indent for everything below. Otherwise it looks perfect now!